AdyenDropIn/Components/PreApplePayComponent.swift

```diff
@@ -98,7 +95,35 @@ extension PreApplePayComponent: PreApplePayViewDelegate {
     
     /// :nodoc:
     internal func pay() {
-        presentationDelegate?.present(component: applePayComponent)
+        let component: ApplePayComponent
+        if let applePayComponent = applePayComponent {
+            component = applePayComponent
+        } else {
+            do {
+                component = try createApplePayComponent()
+                applePayComponent = component
+            } catch {
+                delegate?.didFail(with: error, from: self)
+                return
+            }
+        }
```

Comment:
when clicking pay, maybe the `ApplPayComponent` should be always reinitialized.

Reply:
Problem is - the only way to check if ApplePay is possible - initiate new pkViewController.
So we have to try building new ApplPayComponent in constructor to prevent "Dummy" Apple Pay button on Drop-in that doesn't do anything...

But you right: after first one - every next click on `pay` will reinitialize the ApplPayComponent